src/components/RoomSelection/RoomSelection.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import { Container, Box, Typography, Card, Divider } from "@mui/material";
+import RoomSelectionCards from "./RoomSelectionCards";
+import { iRoomCards } from "../../interfaces/interfaces";
+import { getAllRooms } from "../../api/getRequests";
+export const RoomSelection = () => {
+    const [roomSelectionData, setRoomSelection] = useState<iRoomCards[]>();
+    const getRooms = async () => {
+        try {
+            const data = await getAllRooms();
+            setRoomSelection(data);
+        } catch (error) {
+            console.log(error);
+        }
+    };
+    const mapElements=()=>{
+        return roomSelectionData?.map((e)=>
+        <RoomSelectionCards
+          name={e.name}
+          key={e.id}
+          avaibility={e.avaibility}
+          capacity={e.capacity}
+          description={e.description}
+          id={e.id}
+        ></RoomSelectionCards>
```

Comment:
If you don't have children for a component you don't need to close the tag like this "</RoomSelectionCards>"
You can write: 
`<RoomSelectionCards
          name={e.name}
          key={e.id}
          avaibility={e.avaibility}
          capacity={e.capacity}
          description={e.description}
          id={e.id}
        />`